lib/afterpay/refund.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Afterpay
+  # The Refund object
+  class Refund
+    # Refunds the Payment
+    # @param order_id [String] unique order id returned by afterpay when capturing payment
+    # @param reference [String] the reference for order at merchant end
+    # @param payment_id [String] Unique Payment id at merchant end
+    # @param amount [Money] amount to be refunded
+    # @return [Refund] the Refund object
+    def self.execute(order_id:, reference:, payment_id:, amount:)
+      request = Afterpay.client.post("/v1/payments/#{ order_id }/refund") do |req|
+        req.body = {
+          requestId: payment_id,
+          merchantRefernce: reference,
+          amount: Utils::Money.api_hash(amount)
+        }
+      end
+
+      new(request.body)
+    end
+
+    attr_accessor :id, :status, :created, :total, :order, :error
```

Comment:
where are you using all these attributes? Please omit the ones you are not using